les/server_handler.go

```diff
@@ -322,7 +322,9 @@ func (h *serverHandler) handleMsg(p *clientPeer, wg *sync.WaitGroup) error {
 						origin = h.blockchain.GetHeaderByNumber(query.Origin.Number)
 					}
 					if origin == nil {
-						atomic.AddUint32(&p.invalidCount, 1)
+						p.invalidLock.Lock()
+						p.invalidCount.Add(1, mclock.Now())
```

Comment:
You have 10 instances of this, maybe it would be nicer to make it a function of the peer (especially because it uses a mutex now).

Reply:
Done